lib/rules/selector-max-specificity/index.mjs

```diff
@@ -80,148 +57,46 @@ const rule = (primary, secondaryOptions) => {
-			// so use `node.value` instead
-			const ownValue = node.value.toLowerCase();
-
-			if (ownValue === ':where') {
-				return zeroSpecificity();
-			}
-
-			let ownSpecificity;
-
-			if (isSelectorIgnored(ownValue)) {
-				ownSpecificity = zeroSpecificity();
-			} else if (aNPlusBOfSNotationPseudoClasses.has(ownValue.replace(/^:/, ''))) {
-				return selectorSpecificity(ignoreOfSelectorIfAny(node));
-			} else {
-				ownSpecificity = selectorSpecificity(node.clone({ nodes: [] }));
-			}
-
-			return specificitySum([ownSpecificity, maxChildSpecificity(node)]);
-		};
-
-		/**
-		 * @param {import('postcss-selector-parser').Node} node
-		 * @returns {boolean}
-		 */
-		const shouldSkipPseudoClassArgument = (node) => {
-			// postcss-selector-parser includes the arguments to nth-child() functions
-			// as "tags", so we need to ignore them ourselves.
-			// The fake-tag's "parent" is actually a selector node, whose parent
-			// should be the :nth-child pseudo node.
-			const parentNode = node.parent && node.parent.parent;
-
-			if (parentNode && parentNode.type === 'pseudo' && parentNode.value) {
-				const pseudoClass = parentNode.value.toLowerCase().replace(/^:/, '');
-
-				return (
-					aNPlusBNotationPseudoClasses.has(pseudoClass) || linguisticPseudoClasses.has(pseudoClass)
-				);
-			}
-
-			return false;
-		};
-
-		/**
-		 * Calculate the specificity of a node parsed by `postcss-selector-parser`.
-		 *
-		 * @param {import('postcss-selector-parser').Node} node
-		 * @returns {Specificity}
-		 */
-		const nodeSpecificity = (node) => {
-			if (shouldSkipPseudoClassArgument(node)) {
-				return zeroSpecificity();
-			}
-
-			switch (node.type) {
-				case 'attribute':
-				case 'class':
-				case 'id':
-				case 'tag':
-					return simpleSpecificity(node);
-				case 'pseudo':
-					return pseudoSpecificity(node);
-				case 'selector':
-					// Calculate the sum of all the direct children
-					return specificitySum(node.map((n) => nodeSpecificity(n)));
-				default:
-					return zeroSpecificity();
-			}
-		};
+		/** @type {import('@csstools/selector-specificity').CustomSpecificityCallback | undefined} */
+		const customSpecificity = secondaryOptions?.ignoreSelectors
+			? (node) => {
+					switch (node.type) {
+						case 'attribute':
+						case 'class':
+						case 'id':
+						case 'tag':
+							if (!isSelectorIgnored(node.toString())) {
+								return;
+							}
+
+							return zeroSpecificity();
+						case 'pseudo': {
+							if (!isSelectorIgnored(node.value.toLowerCase())) {
+								return;
+							}
+
+							if (!node.nodes?.length) {
+								return zeroSpecificity();
+							}
+
+							// We only ignore the current pseudo-class, not the specificity of the child nodes.
+							// Calculate the diff between specificity with and without child nodes.
+							const entireSpecificity = selectorSpecificity(node);
+
+							node.nodes = [];
+							const emptySpecificity = selectorSpecificity(node);
```

Comment:
[ask] If it was necessary to respect the original code, we would clone the node instead of mutating it like this:

```suggestion
							const emptySpecificity = selectorSpecificity(node.clone({ nodes: [] }));
```

This mutation is fine, isn't it?

Reply:
Good idea!

Theoretically it's possible to have an auto fix for this rule with `:where()`
Mutating the selector AST without cloning could be surprising in the future.